tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitIntegration.cs

```diff
@@ -209,5 +210,72 @@ internal static void FinishSkippedScope(Scope scope, string skipReason)
                 }
             }
         }
+
+        internal static bool ShouldSkip(ITest currentTest)
+        {
+            if (!CIVisibility.Settings.IntelligentTestRunnerEnabled)
+            {
+                return false;
+            }
+
+            MethodInfo testMethod = currentTest.Method.MethodInfo;
+            string testName = testMethod.Name;
+            string testSuite = testMethod.DeclaringType?.FullName ?? string.Empty;
+
+            var currentContext = SynchronizationContext.Current;
+            try
+            {
+                SynchronizationContext.SetSynchronizationContext(null);
+                var skippeableTests = CIVisibility.GetSkippableTestsFromSuiteAndNameAsync(testSuite, testName).GetAwaiter().GetResult();
+                if (skippeableTests.Count > 0)
+                {
+                    object[] testMethodArguments = currentTest.Arguments;
+                    ParameterInfo[]? methodParameters = null;
+                    foreach (var skippeableTest in skippeableTests)
+                    {
+                        var parameters = skippeableTest.GetParameters();
+
+                        // Same test name and no parameters
+                        if ((parameters?.Arguments is null || parameters?.Arguments.Count == 0) &&
+                            (testMethodArguments is null || testMethodArguments.Length == 0))
+                        {
+                            return true;
+                        }
+
+                        if (parameters?.Arguments is not null)
+                        {
+                            methodParameters ??= testMethod.GetParameters();
+                            bool matchSignature = true;
+                            for (var i = 0; i < methodParameters.Length; i++)
+                            {
+                                var targetValue = "(default)";
+                                if (i < testMethodArguments.Length)
+                                {
+                                    targetValue = Common.GetParametersValueData(testMethodArguments[i]);
+                                }
+
+                                if (!parameters.Arguments.TryGetValue(methodParameters[i].Name, out var argValue) ||
+                                    (string)argValue != targetValue)
+                                {
+                                    matchSignature = false;
+                                    break;
+                                }
+                            }
+
+                            if (matchSignature)
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                SynchronizationContext.SetSynchronizationContext(currentContext);
+            }
```

Comment:
This code is duplicated between XUnit and NUnit with only one difference: how you get the `testMethodArguments`. Maybe it could be extracted into a function and reused?

Reply:
true, done in https://github.com/DataDog/dd-trace-dotnet/pull/3075/commits/e36e77efc6038afea30a9f22aaeacc4631c560c2